webpack/scenes/ContentViews/Details/Filters/ArtifactsWithNoErrata.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Checkbox } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { editCVFilter } from '../ContentViewDetailActions';
+
+export const ArtifactsWithNoErrataRenderer = ({ filterDetails }) => {
+  const dispatch = useDispatch();
+  const { id, inclusion, type } = filterDetails;
+
+  const artifactAttribute = (type === 'modulemd') ? 'original_module_streams' : 'original_packages';
+  const [artifactsNoErrata, enableArtifactsNoErrata] =
+      useState(filterDetails[artifactAttribute] === true);
+
+  const [isLoading, setLoading] = useState(false);
+
+  const setArtifactsNoErrata = (checked) => {
+    enableArtifactsNoErrata(checked);
+    setLoading(true);
+    dispatch(editCVFilter(id, { [artifactAttribute]: checked }, () => setLoading(false)));
+  };
+
```

Comment:
switch (true) {
case type ==='modulemd' && inclusion: return blah 
case type ==='modulemd' && !inclusion: return blah 
case  inclusion: return blah 
default return blah
} 

The above is a pattern I often use to clean up this kind of logic. But it's very much an optional change if you like it.